crates/mem-pool/src/deposit.rs

```diff
@@ -167,15 +183,27 @@ fn check_deposit_cell(ctx: &RollupContext, cell: &DepositInfo) -> Result<()> {
                 .collect(),
         );
 
-        if let Err(err) = registry_ctx.extract_registry_address_from_deposit(
+        match registry_ctx.extract_registry_address_from_deposit(
             cell.request.registry_id().unpack(),
             &script.code_hash(),
             &script.args().raw_data(),
         ) {
-            return Err(anyhow!(
-                "Failed to extract registry address from deposit, err: {}",
-                err
-            ));
+            Ok(reg_addr) => {
+                //Registry address could be duplicated with a contract account.
+                if let Some(script_hash) = state.get_script_hash_by_registry_address(&reg_addr)? {
+                    if script.hash() != script_hash.as_slice() {
+                        return Err(anyhow!(
+                            "RegistryAddress has been mapped with another script hash before."
```

Comment:
```suggestion
                            "The RegistryAddress {:?} was already occupied by script_hash {}, depositing script_hash: {}"
                            reg_addr,
                            hex::encode(&script_hash().as_bytes()),
                            hex::encode(&script.hash().as_bytes()),
```

Just print more information.